app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -120,25 +123,41 @@ class SchedulePageView @JvmOverloads constructor(
     }
 
     fun updateWith(schedule: Schedule, onEventClicked: (Event) -> Unit) {
-        viewPagerAdapter.updateWith(schedule.pages, onEventClicked)
+        val initialEventForPage = schedule.pages.map { schedule.findNextEventForPage(it, currentTime) }.toTypedArray()
+        viewPagerAdapter.updateWith(schedule.pages, initialEventForPage, onEventClicked)
+
+        val todayPageIndex = schedule.findTodayIndexOrDefault(currentTime)
+        viewpager.setCurrentItem(todayPageIndex, false)
+
+        tabstrip.addOnTabSelectedListener(ScrollingOnTabSelectedListener(schedule, viewPagerAdapter))
         progressbar.visibility = View.GONE
     }
 
-    private class TrackingOnTabSelectedListener constructor(
-            private val analytics: Analytics,
+    private interface OnTabSelectedListener : TabLayout.OnTabSelectedListener {
+
+        override fun onTabReselected(tab: TabLayout.Tab) {}
+        override fun onTabUnselected(tab: TabLayout.Tab) {}
+        override fun onTabSelected(tab: TabLayout.Tab) {}
+    }
+
+    private inner class ScrollingOnTabSelectedListener(
```

Comment:
I might be missing something but is there any reason why the inner qualifier is required?

Reply:
Because `currentTime` currently lives inside the outer class. We can pass it it as a dependency!